homeassistant/components/google_assistant/trait.py

```diff
@@ -869,28 +926,35 @@ async def _execute_vacuum(self, command, data, params, challenge):
                     context=data.context,
                 )
 
-    async def _execute_cover(self, command, data, params, challenge):
+    async def _execute_cover_or_valve(self, command, data, params, challenge):
         """Execute a StartStop command."""
+        domain = self.state.domain
         if command == COMMAND_STARTSTOP:
             if params["start"] is False:
-                if self.state.state in (
-                    cover.STATE_CLOSING,
-                    cover.STATE_OPENING,
-                ) or self.state.attributes.get(ATTR_ASSUMED_STATE):
+                if (
+                    self.state.state
+                    in (
+                        COVER_VALVE_STATES[domain]["closing"],
+                        COVER_VALVE_STATES[domain]["opening"],
+                    )
+                    or domain == valve.DOMAIN
```

Comment:
We want the stop function to be always available since we can not read determine from, the state if `stop` should be available,

Reply:
Why is that?

Reply:
It will ensure that the `Stop` button is shown, not the `Start` button. See also the comment below.